### PR TITLE
chore: add native-image-shared-config as parent

### PR DESCRIPTION
**Pull request by mpeddada1**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/google-http-java-client/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

If you write sample code, please follow the [samples format](
https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md).

### CODE REVIEW

File: .kokoro/presubmit/graalvm-native-a.cfg [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # Configure the docker image for kokoro-trampoline.
 env_vars: {
   key: "TRAMPOLINE_IMAGE"
-  value: "gcr.io/cloud-devrel-kokoro-resources/graalvm:22.3.3"
+  value: "gcr.io/cloud-devrel-public-resources/graalvm_a:1.7.1"
```
Comment (burkedavison) [inline .kokoro/presubmit/graalvm-native-a.cfg:6]: Should these `graalvm-native-x.cfg`s have a dependabot/renovatebot comment to tie the version to native-image-shared-config?

Comment (mpeddada1) [reply .kokoro/presubmit/graalvm-native-a.cfg:6]: Good point, enabled renovate-bot to update the java-shared-config version. Tested at https://github.com/mpeddada1/forking-renovate-repro/pull/8

File: renovate.json [hunk 2]
```diff
@@ -11,6 +11,17 @@
...
   ],
   "ignorePaths": [".kokoro/requirements.txt"],
+  "customManagers": [
+    {
+      "customType": "regex",
+      "fileMatch": [
+        "^.kokoro/presubmit/graalvm-native.*.cfg$"
+      ],
+      "matchStrings": ["value: \"gcr.io/cloud-devrel-public-resources/graalvm.*:(?<currentValue>.*?)\""],
+      "depNameTemplate": "com.google.cloud:google-cloud-shared-config",
```
Comment (burkedavison) [inline renovate.json:21]: Does this repo know about `google-cloud-shared-config`? Or should this be `native-image-shared-config`?

Comment (mpeddada1) [reply renovate.json]: Good catch. For consistency, it makes sense for this to use native-image-shared-config.

**Review by burkedavison — APPROVED** [review]

**Pull request merged**